run.py

```diff
@@ -1011,6 +1011,17 @@ def main():
     # Sort alphabetically.
     tests.sort()
 
+    # If we don't find any tests, let's let the user know and exit
+    if not tests:
+        # Let's make some space between our warnings and other possible output
+        print()
+        if args.testdir:
+            print("Warning: Couldn't find any directory to match the pattern \"%s\"" % args.testdir)
+        else:
+            for pattern in args.patterns:
+                print("Warning: Couldn't find any tests to match the pattern \"%s\"" % pattern.rstrip(','))
```

Comment:
Looks great to me. Will wait for others to see if they like the output.
One q: why are we using `.rstrip(',')` here?

Reply:
> One q: why are we using `.rstrip(',')` here?

that's probably because I added them to my list of arguments 😅 but now that I'm checking, I think that was just bad user behavior. I'm ALSO thinking that... if we have more than one pattern and one matches but the other doesn't, this code isn't enough to cover that corner case.

So I guess we have incoming v4 :x